s2a/src/test/java/io/grpc/s2a/handshaker/IntegrationTest.java

```diff
@@ -0,0 +1,322 @@
+        + "MEAwDgYDVR0PAQH/BAQDAgGGMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYEFMX+\n"
+        + "vebuj/lYfYEC23IA8HoIW0HsMAoGCCqGSM49BAMCA0kAMEYCIQDETd27nsUTXKWY\n"
+        + "CiOno78O09gK95NoTkPU5e2chJYMqAIhALYFAyh7PU5xgFQsN9hiqgsHUc5/pmBG\n"
+        + "BGjJ1iz8rWGJ\n"
+        + "-----END CERTIFICATE-----";
+  private static final String PRIVATE_KEY =
+      "-----BEGIN PRIVATE KEY-----\n"
+        + "MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgqA2U0ld1OOHLMXWf\n"
+        + "uyN4GSaqhhudEIaKkll3rdIq0M+hRANCAAQGFlJpLxJMh4HuUm0DKjnUF7larH3t\n"
+        + "JvroQ12xpk+pPKQepn4ILoq9lZ8Xd3jzU98eDRXG5f4VjnX98DDHE4Id\n"
+        + "-----END PRIVATE KEY-----";
+
+  private String s2aAddress;
+  private int s2aPort;
+  private Server s2aServer;
+  private String s2aDelayAddress;
+  private int s2aDelayPort;
+  private Server s2aDelayServer;
+  private String mtlsS2AAddress;
+  private int mtlsS2APort;
+  private Server mtlsS2AServer;
+  private int serverPort;
+  private String serverAddress;
+  private Server server;
+
+  @BeforeClass
+  public static void setUpClass() {
+    System.setProperty("GRPC_EXPERIMENTAL_ENABLE_NEW_PICK_FIRST", "false");
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    s2aPort = Utils.pickUnusedPort();
+    s2aAddress = "localhost:" + s2aPort;
+    s2aServer = ServerBuilder.forPort(s2aPort).addService(new FakeS2AServer()).build();
+    logger.info("S2A service listening on localhost:" + s2aPort);
+    s2aServer.start();
+
+    mtlsS2APort = Utils.pickUnusedPort();
+    mtlsS2AAddress = "localhost:" + mtlsS2APort;
+    File s2aCert = new File("src/test/resources/server_cert.pem");
+    File s2aKey = new File("src/test/resources/server_key.pem");
+    File rootCert = new File("src/test/resources/root_cert.pem");
+    ServerCredentials s2aCreds =
+        TlsServerCredentials.newBuilder()
+            .keyManager(s2aCert, s2aKey)
+            .trustManager(rootCert)
+            .clientAuth(TlsServerCredentials.ClientAuth.REQUIRE)
+            .build();
+    mtlsS2AServer =
+        NettyServerBuilder.forPort(mtlsS2APort, s2aCreds).addService(new FakeS2AServer()).build();
+    logger.info("mTLS S2A service listening on localhost:" + mtlsS2APort);
+    mtlsS2AServer.start();
+
+    s2aDelayPort = Utils.pickUnusedPort();
+    s2aDelayAddress = "localhost:" + s2aDelayPort;
+    s2aDelayServer = ServerBuilder.forPort(s2aDelayPort).addService(new FakeS2AServer()).build();
+
+    serverPort = Utils.pickUnusedPort();
+    serverAddress = "localhost:" + serverPort;
+    server =
+        NettyServerBuilder.forPort(serverPort)
+            .addService(new SimpleServiceImpl())
+            .sslContext(buildSslContext())
+            .build();
+    logger.info("Simple Service listening on localhost:" + serverPort);
+    server.start();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    server.shutdown();
+    s2aServer.shutdown();
+    s2aDelayServer.shutdown();
+    mtlsS2AServer.shutdown();
```

Comment:
Change this to `mtlsS2AServer.shutdown(). awaitTermination(10, TimeUnit.SECONDS)`

Reply:
Thanks Larry! Done in https://github.com/grpc/grpc-java/pull/11113/commits/752627a8e5b383f9bb06c5296bd1691a5987f1da